MySQL does not start with overlay2 on docker for OSX

On docker for OSX sometimes there is a problem with the SQL server failing to start.
I look like it is because the OverlayFS only implements a subset of the POSIX standards.

A workaround is to touch the mysql files before starting mysql

**find /var/lib/mysql -type f -exec touch {} \; && service mysql start**

[https://github.com/docker/for-linux/issues/72](https://github.com/docker/for-linux/issues/72)

`# cat /var/log/mysql/error.log`
_[ERROR] Fatal error: Can't open and lock privilege tables: Got error 140 from storage engine_

--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -17,6 +17,14 @@ if [ -z $KEYSTONE_DB_HOST ]; then
     KEYSTONE_DB_HOST=localhost
     # start mysql locally
     service mysql restart
+
+    #Docker OverlayFS compatibility: implements subset POSIX standards
+    #https://docs.docker.com/storage/storagedriver/overlayfs-driver/
+    #ALT: attach /var/lib/mysql as a volume to avoid timeout
+    if [ $? ] ; then
+        find /var/lib/mysql -type f -exec touch {} \;
+        service mysql restart
+    fi
 else
     if [ -z $KEYSTONE_DB_ROOT_PASSWD_IF_REMOTED ]; then
         echo "Your'are using Remote MySQL Database; "

